llvm/lib/CheerpUtils/TypeOptimizer.cpp

```diff
@@ -1166,6 +1166,19 @@ Function* TypeOptimizer::rewriteIntrinsic(Function* F, FunctionType* FT)
 	return newF;
 }
 
+Value* TypeOptimizer::getConstantForGEP(Type* Int32Ty, uint32_t constant, Instruction* GEP)
```

Comment:
isn't accepting targetType better here, instead of getting the type from GEP?
If GEP is null we probably still want to return a vector if appropriate

Reply:
targetType is not necessarily a vector type when the return type of the GEP is. In this specific example, targetType is [16 x i8], arrayType.

Reply:
Addressed this by sending an extra parameter to rewriteGEPIndexes, the returnType.